src/modules/deal-investor/deal-investor.service.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable } from '@nestjs/common';
+import { DateStringTransformations, UkefId } from '@ukef/helpers';
+import { AcbsAuthenticationService } from '@ukef/modules/acbs/acbs-authentication.service';
+import { AcbsDealPartyService } from '@ukef/modules/acbs/acbs-deal-party.service';
+import { AcbsResourceNotFoundException } from '@ukef/modules/acbs/exception/acbs-resource-not-found.exception';
+
+import { DealInvestor } from './deal-investor.interface';
+
+@Injectable()
+export class DealInvestorService {
+  // TODO: make DateStringTransformations injectable, maybe it needs to be part of module for this to work.
+  private readonly dateStringTransformations: DateStringTransformations = new DateStringTransformations();
+
+  constructor(private readonly acbsAuthenticationService: AcbsAuthenticationService, private readonly acbsDealPartyService: AcbsDealPartyService) {}
+
+  async getDealInvestors(dealIdentifier: UkefId): Promise<DealInvestor[]> {
+    const idToken = await this.acbsAuthenticationService.getIdToken();
+    // TODO: move E1 to ACBS default values when they are available.
+    const portfolio = 'E1';
+    const investorsInAcbs = await this.acbsDealPartyService.getDealPartyForDeal(portfolio, dealIdentifier, idToken);
+    if (investorsInAcbs === null) {
```

Comment:
should we use here:
```js
if (!investorInAcbs) {...}
```
this because it covers both `null` and `undefined`

Reply:
Makes sense, made it more generic.